### DISCUSSION TITLE
CSS files not included with SSR:

### DISCUSSION THREAD

**Discussion by ScottPierce** [vikejs/vike#995] [Discussion] [answered]
I have a very simple example that I'm trying to get working with SSR:

index.page.server.tsx:
```
import {PageContext} from "../../../renderer/types";
import './blog.css'

export function Page(context: PageContext) {
    // @ts-ignore
    const id = context.routeParams?.id

    return (
        <>
            <h1 className={(id == '1') ? "a" : "b"}>Blog ID {id} {new Date().toISOString()}</h1>
        </>
    )
}
```

blog.css:
```
.a {
    color: red;
}

.b {
    color: blue;
}
```

it works using `yarn dev`, but the css isn't loaded when using `yarn prod`.

It seems to work if I change `index.page.server.tsx` to `index.page.tsx`, however this is a purely static website.

What am I doing wrong?

**Comment by brillout** [comment, marked as answer]
See https://vite-plugin-ssr.com/includeAssetsImportedByServer.

**Reply by brillout** [reply]
The flag is now stable and `true` by default https://github.com/brillout/vite-plugin-ssr/commit/67aa8d4cf3edd474c52b45066ffa1affeb62338f.

**Comment by ScottPierce** [comment]
@brillout that fixed it. Will this work for something like tailwind, and optimize away the unused CSS for the page?

**Reply by brillout** [reply]
> optimize away the unused CSS

Vite doesn't do that. It does, however, split the CSS (as you probably already expect).

**Reply by ScottPierce** [reply]
Any recommendations for where I can read about how it splits it out and decides? I'm pretty new to vite.

**Reply by brillout** [reply]
That's how it basically works: it only loads code (JS and CSS) that is needed for a page (transitive imports of your `.page.`/`+` files).

**Reply by brillout** [reply]
Btw. in case you're able to convince your employer: https://github.com/sponsors/brillout &mdash; it makes a big difference.

**Reply by ScottPierce** [reply]
@brillout We're in the process of considering this project for a fairly large move of our template rendering engine. If we end up using it, we will probably support you. We do something similar for https://github.com/graphile/postgraphile

**Reply by brillout** [reply]
Sounds good.

> considering this project

If you have questions, don't hesitate to ask here on GitHub Discussions. (And feel free to PM me on Discord for non-technical questions.)

Also let me know if you hit a blocker, as the intention of [the `v1.0.0` release](https://github.com/brillout/vite-plugin-ssr/labels/v1%20release%20%3Astar2%3A) is to make it a feature complete, highly polished, and robust tool to build modern apps.

Since you seem to be using React, I recommend setting [`ssr.noExternal`](https://vitejs.dev/config/ssr-options.html#ssr-noexternal) to `true`. The upside is that it removes [issues around wrongly published React libraries](https://github.com/brillout/vite-plugin-ssr/issues/984). The downside is that it makes development slightly slower (but it's definitely worth it). This is a new approach I'll be making mainstream (vite-plugin-ssr will show a warning to React users not setting `ssr.noExternal` to `true`). The development speed degradation should be fairly unnoticeable. (Vite can transpile `node_modules/` files very quickly as it can use the fast transpiler [esbuild](https://esbuild.github.io/) without needing slow user-land transpilers such as Babel.)

**Reply by ScottPierce** [reply]
Thank you, we've set this based on your recommendation:
```
    ssr: {
        noExternal: ['react', 'react-dom']
    }
```

Thank you.
